apps/system/js/app_window.js

```diff
@@ -227,10 +227,17 @@
     if (!this.isBrowser()) {
       return false;
     }
-
-    var url = new URL(this.config.url);
-    var path = url.hostname + url.pathname;
-    return path.indexOf(scope) === 0;
+    // within-scope per http://www.w3.org/TR/appmanifest/#dfn-within-scope
+    // except we also support paths
+    var withinScope;
+    var target = this.config.url;
+    if (scope) {
+      var scopeURL = new URL(scope);
+      withinScope = target.indexOf(scopeURL.origin + scopeURL.pathname) === 0;
```

Comment:
I still don't get this. The scope passed by parameter is the one we want to check. We don't need to add or remove any path. target.indexOf(scope) === 0 should be ok.